include/podio/UserDataCollection.h

```diff
@@ -31,26 +32,27 @@ using SupportedUserDataTypes =
     std::tuple<float, double, int8_t, int16_t, int32_t, int64_t, uint8_t, uint16_t, uint32_t, uint64_t>;
 
 /**
- * Alias template to be used to enable template specializations only for the types listed in the
- * SupportedUserDataTypes list
+ * Alias template to be used to enable template specializations only for the
+ * types listed in the SupportedUserDataTypes list or if they are components
+ * that were generated by podio
  */
 template <typename T>
-using EnableIfSupportedUserType = std::enable_if_t<detail::isInTuple<T, SupportedUserDataTypes>>;
+using EnableIfSupportedUserType =
+    std::enable_if_t<detail::isInTuple<T, SupportedUserDataTypes> || detail::isComponent<T>>;
```

Comment:
Any chance this could be exposed as a public compile-time boolean? I can use the `detail` namespace of course, but would be nice to use public API only.

Reply:
Ideally the whole expression. The type tuple is already public so with my own `isInTuple` I can use it, but there's no good way to check if a type is a component from the outside.

Reply:
Done. We also do the same in our `GenericParameters` so I don't see a reason to not do this here as well. Plus in this way we can keep the `isComponent` bit in `detail` and not expose that directly.